### PR TITLE
EDUCATOR-2231 remove course run enrollment dates references

### PR DESCRIPTION
**Pull request by Rabia23**
### Description
This PR removes all the reference to the enrollment dates columns.

Related edx-platform PR: https://github.com/edx/edx-platform/pull/17559

### Reviewers
If you've been tagged for review, please check your corresponding box once you've given the :+1:.
- [ ]  @schenedx 
- [ ]  @awaisdar001 

### Post-review
- [ ] Rebase and squash commits

### CODE REVIEW

File: course_discovery/apps/publisher/api/v1/tests/test_views.py [hunk 1]
```diff
@@ -148,8 +148,6 @@ def test_publish(self, mock_access_token):  # pylint: disable=unused-argument,to
         assert discovery_course_run.full_description_override is None
         assert discovery_course_run.start == publisher_course_run.start
         assert discovery_course_run.end == publisher_course_run.end
-        assert discovery_course_run.enrollment_start == publisher_course_run.enrollment_start
```
Comment (awaisdar001) [inline course_discovery/apps/publisher/api/v1/tests/test_views.py:148]: Test to see if `discovery_course_run.enrollment_start == None`

**Review by awaisdar001 — CHANGES_REQUESTED** [review]

File: course_discovery/apps/publisher/api/v1/tests/test_views.py [hunk 2]
```diff
@@ -148,8 +148,8 @@ def test_publish(self, mock_access_token):  # pylint: disable=unused-argument,to
...
-        assert discovery_course_run.enrollment_end == publisher_course_run.enrollment_end
+        assert discovery_course_run.enrollment_start is None
+        assert discovery_course_run.enrollment_end is None
         assert discovery_course_run.pacing_type == publisher_course_run.pacing_type
         assert discovery_course_run.min_effort == publisher_course_run.min_effort
         assert discovery_course_run.max_effort == publisher_course_run.max_effort
```
Comment (awaisdar001) [inline course_discovery/apps/publisher/api/v1/tests/test_views.py:152]: Just curious to know, can we use `self.assertEqual` here?

**Review by awaisdar001 — COMMENTED** [review]
👍

**Comment by Rabia23** [comment]
@schenedx Can you please have a look at this one too?

**Review by schenedx — COMMENTED** [review]
I believe you can also remove the reference of enrollment_start and end from the model.py, just don't check in the migration file generated.
Additionally, why update all the assert styles in tests? What is the value of making those changes?

**Comment by Rabia23** [comment]
@schenedx just curious to know, what is the hesitation of not generating migration file after removing the reference of enrollment_start and end from the model.py file.

**Comment by Rabia23** [comment]
`assert` is pure python related thing and `self. assertEqual()` comes from `unittest.TestCase.assertEqual()` so i think it's good idea to use `assertEqual` instead of `assert` in unit tests.

**Comment by schenedx** [comment]
@Rabia23 
Do you know how the code deploys to production? Do you know that data base table change will be first introduced into the production system, then after a few minutes, the actual server code will change? If we introduce migration within the same PR, the database will have breaking change deployed before the servers code will be updated. Within that few minutes a lot of things will break. If you still have questions, I strongly recommend you talk to someone like @waheedahmed to understand more.
As for the assert changes, there are school of thought that python asserts are better choice than the unit test implementation of the asserts. If the changes from one thing to the other do not have any value, why make them? I would be against changing code because you thought one way is better than the other way while both ways works.

**Comment by Rabia23** [comment]
@schenedx Thank you for taking deployment process into my account . I didn't know this before. I messed up this PR so i have opened new PR https://github.com/edx/course-discovery/pull/1350 in which I have removed the reference of `enrollment_start` and `enrollment_end` from the model.py file and also revert back `assert` changes. 
Please have a look at that PR.
Thanks.

**Pull request closed without merging**